pyteal/ast/abi/address.py

```diff
@@ -93,21 +90,9 @@ def set(
 
         match value:
             case ComputedValue():
-                pts = value.produced_type_spec()
-                if pts == AddressTypeSpec() or pts == StaticArrayTypeSpec(
-                    ByteTypeSpec(), AddressLength.Bytes
-                ):
-                    return value.store_into(self)
-
-                raise TealInputError(
-                    f"Got ComputedValue with type spec {pts}, expected AddressTypeSpec or StaticArray[Byte, Literal[AddressLength.Bytes]]"
-                )
+                return self._set_with_computed_type(value)
             case BaseType():
-                if (
-                    value.type_spec() == AddressTypeSpec()
-                    or value.type_spec()
-                    == StaticArrayTypeSpec(ByteTypeSpec(), AddressLength.Bytes)
-                ):
+                if type_spec_is_assignable_to(value.type_spec(), AddressTypeSpec()):
                     return self.stored_value.store(value.stored_value.load())
 
```

Comment:
@ahangsu Unrelated to the line - Based on the PR's changes, the `TealInputError` at the end of the method should remove _StaticArray_ from its message.  

Not requesting an immediate change because there's more pressing threads open.  I leave the question to avoid forgetting the concern prior to merge.